Assets/Scripts/UI/PickupPup/DogProfile/DogProfileButtonController.cs

```diff
@@ -0,0 +1,109 @@
+﻿/*
+ * Author: Grace Barrett-Snyder 
+ * Description: Controls navigation between Dog Profiles
+ */
+
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DogProfileButtonController : PPUIButtonController
+{
+	public MonoActionInt OnSwitchProfile;
+
+    #region Instance Accessors
+
+    public bool IsInitialized
+    {
+        get;
+        private set;
+    }
+
+    #endregion
+
+    [SerializeField]
+    UIButton pageBackwardButton;
+    [SerializeField]
+    UIButton pageForwardButton;
+
+    DogProfile parentWindow;
+    ToggleableColorUIButton[] pageButtons;
+    ToggleableColorUIButton selectedPageButton;
+
+    List<Dog> dogsList;
+    int currentProfileIndex;
+
+    #region MonoBehaviourExtended Overrides
+
+    protected override void setReferences()
+    {
+        base.setReferences();
+        // GetComponentInParent also checks the current GameObject (as per the current prefab)
+        parentWindow = GetComponentInParent<DogProfile>();
+    }
+
+    #endregion
+
+	public void Init(DogProfile profile, DogDescriptor[] dogs)
+    {
+        this.parentWindow = profile;
+        IsInitialized = true;
+
+		updateDogList (dogs);
+
+		pageBackwardButton.SubscribeToClick(previousProfile);
+		pageForwardButton.SubscribeToClick(nextProfile);
+    }
+
+	void updateDogList(DogDescriptor[] dogInfos) {
+		var adoptedDogs = PPDataController.GetInstance.AdoptedDogs;
+		DogFactory dogFactory = new DogFactory(hideGameObjects: true);
+		dogsList = dogFactory.CreateGroupList(new List<DogDescriptor>(dogInfos));
+	}
+
+    void switchToProfile(int index)
+    {
+        checkReferences();
+        currentProfileIndex = index;
+        checkCurrentIndex();
+        parentWindow.SetProfile(dogsList[currentProfileIndex]);
+		if (OnSwitchProfile != null)
```

Comment:
- `if` statements should always including brackets in our codebase
- no space between paren and if